src/datasets/transformations.jl

```diff
@@ -157,6 +161,36 @@ getobs(jdata, 15) == 15
 """
 joinobs(datas...) = JoinedData(datas)
 
+"""
+    tokenize(type, input)
+    type = :words or :chars
+
+Tokenizes an input string or stream into pieces depending on selected type. 
+"""
+
+struct Tokenizer
+    data::AbstractArray{AbstractString}
+end
```

Comment:
For performance reasons, in Julia we use type parameters to get concrete typed fields:
```suggestion
struct Tokenizer{T<:AbstractArray{<:AbstractString}}
    data::T
end
```
Also, should this be a `AbstractVector` or `AbstractArray`?

Reply:
I chose AbstractArray because AbstractVector <: AbstractArray